source/common/upstream/health_checker_impl.cc

```diff
@@ -199,6 +199,7 @@ void HttpHealthCheckerImpl::HttpActiveHealthCheckSession::onResponseComplete() {
   if (isHealthCheckSucceeded()) {
     handleSuccess();
   } else {
+    host_->failureTypeFlagSet(Host::FailureTypeFlag::UNHEALTHY);
```

Comment:
Ideally these are directly adjacent to where we transition the `healthy()` status, this will make it easier to ready if possible.